central/graphql/resolvers/image_vulnerabilities_test.go

```diff
@@ -0,0 +1,364 @@
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	image := s.getImageResolver(ctx, "sha1")
+
+	vulns, err := image.ImageVulnerabilities(ctx, PaginatedQuery{})
+	s.NoError(err)
+	s.Equal(3, len(vulns))
+
+	image = s.getImageResolver(ctx, "sha2")
+
+	vulns, err = image.ImageVulnerabilities(ctx, PaginatedQuery{})
+	s.NoError(err)
+	s.Equal(5, len(vulns))
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestImageVulnerabilityMiss() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	vulnID := graphql.ID("invalid")
+
+	_, err := s.resolver.ImageVulnerability(ctx, IDQuery{ID: &vulnID})
+	s.Error(err)
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestImageVulnerabilityHit() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	vulnID := graphql.ID("cve-2018-1#")
+
+	vuln, err := s.resolver.ImageVulnerability(ctx, IDQuery{ID: &vulnID})
+	s.NoError(err)
+	s.Equal(vulnID, vuln.ID(ctx))
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestImageVulnerabilityCount() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	count, err := s.resolver.ImageVulnerabilityCount(ctx, RawQuery{})
+	s.NoError(err)
+	s.Equal(int32(5), count)
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestImageVulnerabilityCountScoped() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	image := s.getImageResolver(ctx, "sha1")
+
+	count, err := image.ImageVulnerabilityCount(ctx, RawQuery{})
+	s.NoError(err)
+	s.Equal(int32(3), count)
+
+	image = s.getImageResolver(ctx, "sha2")
+
+	count, err = image.ImageVulnerabilityCount(ctx, RawQuery{})
+	s.NoError(err)
+	s.Equal(int32(5), count)
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestImageVulnerabilityCounter() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	count, err := s.resolver.ImageVulnerabilityCounter(ctx, RawQuery{})
+	s.NoError(err)
+	checkVulnerabilityCounter(s.T(), count, 5, 1, 1, 2, 1, 1)
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestImageVulnerabilityCounterScoped() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	image := s.getImageResolver(ctx, "sha1")
+
+	count, err := image.ImageVulnerabilityCounter(ctx, RawQuery{})
+	s.NoError(err)
+	checkVulnerabilityCounter(s.T(), count, 3, 1, 1, 0, 1, 1)
+
+	image = s.getImageResolver(ctx, "sha2")
+
+	count, err = image.ImageVulnerabilityCounter(ctx, RawQuery{})
+	s.NoError(err)
+	checkVulnerabilityCounter(s.T(), count, 5, 1, 1, 2, 1, 1)
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestTopImageVulnerabilityUnscoped() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	_, err := s.resolver.TopImageVulnerability(ctx, RawQuery{})
+	s.Error(err)
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestTopImageVulnerability() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	image := s.getImageResolver(ctx, "sha1")
+
+	_, err := image.TopImageVulnerability(ctx, RawQuery{})
+	s.NoError(err)
+
+	// TODO figure out how to test this
```

Comment:
verify that the returned vuln is the one with the highest cvss score in that image.

Reply:
It doesn't seem like TopImageVulnerability actually returns data correctly, I've tried tweaking the CVSS score on the CVEs loaded into the datastore and it just always returns the same CVE. That's what the TODO was meant to capture